apis/firestore/v1beta1/database_types.go

```diff
@@ -42,6 +42,51 @@ type FirestoreDatabaseSpec struct {
 	// Whether to enable the PITR feature on this database.
 	// See https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases#pointintimerecoveryenablement for more info.
 	PointInTimeRecoveryEnablement *string `json:"pointInTimeRecoveryEnablement,omitempty"`
+
+	// State of delete protection for the database.
+	// +kcc:proto:field=google.firestore.admin.v1.Database.delete_protection_state
+	DeleteProtectionState *string `json:"deleteProtectionState,omitempty"`
+
+	// NOTYET(PENDING)
+	// // The type of the database.
+	// //  See https://cloud.google.com/datastore/docs/firestore-or-datastore for
+	// //  information about how to choose.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.type
+	// Type *string `json:"type,omitempty"`
+
+	// // The concurrency control mode to use for this database.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.concurrency_mode
+	// ConcurrencyMode *string `json:"concurrencyMode,omitempty"`
+
+	// // Whether to enable the PITR feature on this database.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.point_in_time_recovery_enablement
+	// PointInTimeRecoveryEnablement *string `json:"pointInTimeRecoveryEnablement,omitempty"`
+
+	// // The App Engine integration mode to use for this database.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.app_engine_integration_mode
+	// AppEngineIntegrationMode *string `json:"appEngineIntegrationMode,omitempty"`
+
+	// // Optional. Presence indicates CMEK is enabled for this database.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.cmek_config
+	// CmekConfig *Database_CmekConfig `json:"cmekConfig,omitempty"`
+
+	// // Optional. Input only. Immutable. Tag keys/values directly bound to this
+	// //  resource. For example:
+	// //    "123/environment": "production",
+	// //    "123/costCenter": "marketing"
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.tags
+	// Tags map[string]string `json:"tags,omitempty"`
+
+	// // This checksum is computed by the server based on the value of other
+	// //  fields, and may be sent on update and delete requests to ensure the
+	// //  client has an up-to-date value before proceeding.
+	// // +kcc:proto:field=google.firestore.admin.v1.Database.etag
+	// Etag *string `json:"etag,omitempty"`
+
+	// // Immutable. The edition of the database.
```

Comment:
Nit: Should this be a status field rather than a spec field?

Reply:
Probably, but it's commented out so I don't think it matters for now.

I've added them commented out here because it is very convenient to "grab" these fields when they are first generated, because after that the main type is no longer generated.  So my workflow for a new type is typically to add it generate.sh, run generate.sh, copy the generated fields into spec/observedstate and then comment out the ones I don't want right away (which are _normally_ few, but in this case we're dealing with terraform decisions)